var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -216,7 +216,7 @@ class Cp2k(MakefilePackage, CudaPackage, CMakePackage, ROCmPackage):
         conflicts("~mpi_f08", when="^mpich@4.1:")
 
     with when("+cosma"):
-        depends_on("cosma+scalapack")
+        depends_on("cosma+scalapack+shared")
```

Comment:
Without shared library, cp2k +cosma could still get built through makefile method. I suggest using +shared only with "build_system=cmake", or patching cp2k CMakeLists.txt .
cp2k itself do not rely on any cosma shared library. Using +shared only provides "cosma::cosma_pxgemm" target demanded by cp2k CMakeLists.txt. In fact cp2k does not link against this target; cp2k links against libcosma_prefixed_pxgemm and libcosma, provided by both static and shared version.